packages/nitro-protocol/gas-benchmarks/fixtures.ts

```diff
@@ -21,129 +23,208 @@ import {Bytes} from '../src/contract/types';
+  supportProof(
+    // for concluding
+    state: State
+  ): {
+    largestTurnNum: number;
+    fixedPart: FixedPart;
+    appPartHash: Bytes32;
+    outcomeBytes: Bytes;
+    numStates: 1;
+    whoSignedWhat: number[];
+    sigs: Signature[];
+  } {
+    return {
+      largestTurnNum: state.turnNum,
+      fixedPart: getFixedPart(state),
+      appPartHash: hashAppPart(state),
+      outcomeBytes: encodeOutcome(state.outcome),
+      numStates: 1,
+      whoSignedWhat: this.wallets.map(() => 0),
+      sigs: this.wallets.map(w => signState(state, w.privateKey).signature),
+    };
+  }
+
+  async concludePushOutcomeAndTransferAllTx(assetHolderAddress: string) {
+    const fP = this.supportProof(this.finalState(assetHolderAddress));
+    return await nitroAdjudicator.concludePushOutcomeAndTransferAll(
+      fP.largestTurnNum,
+      fP.fixedPart,
+      fP.appPartHash,
+      fP.outcomeBytes,
+      fP.numStates,
+      fP.whoSignedWhat,
+      fP.sigs
+    );
+  }
 
-export function finalizationProof( // for concluding
-  state: State
-): {
-  largestTurnNum: number;
-  fixedPart: FixedPart;
-  appPartHash: Bytes32;
-  outcomeBytes: Bytes;
-  numStates: 1;
-  whoSignedWhat: [0, 0];
-  sigs: [Signature, Signature];
-} {
-  return {
-    largestTurnNum: state.turnNum,
-    fixedPart: getFixedPart(state),
-    appPartHash: hashAppPart(state),
-    outcomeBytes: encodeOutcome(state.outcome),
-    numStates: 1,
-    whoSignedWhat: [0, 0],
-    sigs: [
-      signState(state, Alice.privateKey).signature,
-      signState(state, Bob.privateKey).signature,
-    ],
-  };
+  async challengeTx(assetHolderAddress: string) {
+    const proof = this.counterSignedSupportProof(this.someState(assetHolderAddress));
+    return await nitroAdjudicator.challenge(
+      proof.fixedPart,
+      proof.largestTurnNum,
+      proof.variableParts,
+      proof.isFinalCount,
+      proof.signatures,
+      proof.whoSignedWhat,
+      proof.challengeSignature
+    );
+  }
 }
 
+// An application channel
+export const X = new TestChannel(
+  2,
+  [Alice, Bob],
+  [
+    {destination: convertAddressToBytes32(Alice.address), amount: '0x5'},
+    {destination: convertAddressToBytes32(Bob.address), amount: '0x5'},
+  ]
+);
+export const Y = new TestChannel(
+  3,
+  [Alice, Bob],
+  [
+    {destination: convertAddressToBytes32(Alice.address), amount: '0x5'},
+    {destination: convertAddressToBytes32(Bob.address), amount: '0x5'},
+  ]
+);
+
+// Ledger funding
+export const LforX = new TestChannel(4, [Alice, Bob], [{destination: X.channelId, amount: '0xa'}]);
+
+// Virtual funding
+export const J = new TestChannel(
+  5,
+  [Alice, Bob, Ingrid],
+  [
+    {destination: X.channelId, amount: '0xa'},
+    {destination: convertAddressToBytes32(Ingrid.address), amount: '0xa'},
+  ]
+);
+export const G = new TestChannel(6, [Alice, Ingrid], {
+  targetChannelId: J.channelId,
+  destinations: [
+    convertAddressToBytes32(Alice.address),
+    convertAddressToBytes32(Bob.address),
```

Comment:
Probably immaterial with respect to the gas cost tests, but shouldn't this guarantor channel be funding [alice, ingrid, X] rather than [alice, bob, X]?

Reply:
Good catch. Fixed 91b0f75 (did cause some small changes to gas).